client/src/components/Ratings&Reviews/AddReview.jsx

```diff
@@ -0,0 +1,479 @@
+import React from 'react';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+
+class AddReview extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.chooseStarRating = this.chooseStarRating.bind(this);
+    this.trackWordCount = this.trackWordCount.bind(this);
+    this.showPhotos = this.showPhotos.bind(this);
+    this.getProductName = this.getProductName.bind(this);
+    this.submitReview = this.submitReview.bind(this);
+    this.chooseCharacteristics = this.chooseCharacteristics.bind(this);
+    this.modalSummary = this.modalSummary.bind(this);
+    this.modalBody = this.modalBody.bind(this);
+    this.modalNickname = this.modalNickname.bind(this);
+    this.modalEmail = this.modalEmail.bind(this);
+    const { productId } = this.props;
+    this.state = {
+      overallStarChosen: 0,
+      newReview: {
+        product_id: productId, rating: 0, summary: '', body: '', recommend: '', name: '', email: '', photos: [], characteristics: {},
+      },
+      bodyWordCount: 0,
+      bodyWordCountCheck: false,
+      photos: [],
+      // urlPhotos: [],
+      productName: '',
+      // error checking
+      ratingError: false,
+      recommendError: false,
+      characteristicsError: false,
+      bodyError: false,
+      nicknameError: false,
+      emailError: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getProductName();
+  }
+
+  // GET product name
+  getProductName() {
+    const { productId } = this.props;
+    $.ajax({
+      url: `/atelier/products/${productId}`,
+      type: 'GET',
+      success: (data) => {
+        this.setState({
+          productName: data.name,
+        });
+      },
+    });
+  }
+
+  // Modal State Updates
+  // Functionality of choosing overall star rating in modal
+  chooseStarRating(e) {
+    const { newReview } = this.state;
+    const overallStar = e.target.id;
+    let num = Number(overallStar[overallStar.length - 1]);
+    let resetStars = 5;
+
+    this.setState({
+      overallStarChosen: Number(overallStar[overallStar.length - 1]),
+    });
+    this.setState({
+      newReview: {
+        ...newReview,
+        rating: num,
+      },
+    });
+    // reset star colors
+    while (resetStars > 0) {
+      const star = $(`#star-${resetStars}`);
+      star.css('color', 'black');
+      resetStars -= 1;
+    }
+    // fill in star colors
+    while (num > 0) {
+      const star = $(`#star-${num}`);
+      star.css('color', 'yellow');
+      num -= 1;
+    }
+  }
+
+  recommendRadio(e) {
+    const { newReview } = this.state;
+    // if yes, true
+    if (e.target.value === 'yes') {
+      this.setState({
+        newReview: {
+          ...newReview,
+          recommend: true,
+        },
+      });
+    }
+    // if no, false
+    if (e.target.value === 'no') {
+      this.setState({
+        newReview: {
+          ...newReview,
+          recommend: false,
+        },
+      });
+    }
```

Comment:
I think this function could be simplified by removing the two if statements and just invoking this.setState where recommend is set to `e.target.value === 'yes' ? true : false`